src/Syntax.hs

```diff
@@ -29,7 +29,7 @@ makeLenses ''Stx
 data ExprF a
   = Id Text
   | String Text
-  | Sig Signal
+  | Lit LiteralInteger
```

Comment:
I'm not a big fan of this constructor name, because strings or other bits we might want can also be fairly characterized as  "literals". I found it a bit confusing at first myself. What about `Int` or `Integer` instead?

Reply:
Changed to `LitInt` to avoid confusion with Haskell's `Integer`, and because it's a tad more descriptive.